### ISSUE TITLE
Implement `MouseEvent` `pageX` and `pageY`

### ISSUE THREAD

**Issue by alexreardon** [jsdom/jsdom#1911]
### Basic info:

- **Node.js version:** 6.9.1
- **jsdom version:** 11.1.0

### Minimal reproduction case

```js
const event = new window.MouseEvent(eventName, {
    bubbles: true,
    cancelable: true,
    view: window,
    pageX,
    pageY,
    button,
  });
```

Currently `event.pageX` and `event.pageY` are not valid properties on a `MouseEvent`. Is there are specific reason they are not supported or should I put together a PR to have this added?

**Comment by domenic** [comment]
What spec are they defined in? Which browsers support them?

**Comment by alexreardon** [comment]
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/pageX) it is defined in the [CSS Object Model (CSSOM) View Module](https://drafts.csswg.org/cssom-view/#dom-mouseevent-pagex) (status: Working Draft)

Using the code below I can confirm that it works in latest Chrome, Safari, Firefox, Edge and IE11.

```js
function log(event) {
  console.log({
    clientX: event.clientX,
    clientY: event.clientY,
    pageX: event.pageX,
    pageY: event.pageY,
  });
}

window.addEventListener('mousemove', log);
```

**Comment by domenic** [comment]
Great, thanks. A pull request would be welcome then. (Note, however, the related bug https://github.com/tmpvar/jsdom/issues/1559 which might get in the way of more advanced scenarios.)

**Comment by benhjames** [comment]
Sorry if this is naïve, but would a PR be as simple as adding `pageX` and `pageY` to here? Would ❤️ to fix this!

https://github.com/jsdom/jsdom/blob/144b39b726eabde92abdfaa362cd68f1d544d138/lib/jsdom/living/events/MouseEvent-impl.js#L10-L29

**Comment by domenic** [comment]
No, it's a decent bit more complicated, as you can see from the spec linked above:

https://drafts.csswg.org/cssom-view/#dom-mouseevent-pagex

**Comment by julienw** [comment]
I think the `Minimal reproduction case` above is incorrect: this is probably illegal to be able to pass these values in the constructor; rather they're computed from clientX/clientY/screenX/screenY.

Other properties are missing, namely `offsetX` `offsetY` `x` and `y`. The 2 latter ones should be straightforward to implement, `offsetX` and `offsetY` are clearly more challenging. They all have a very good support (including IE6, given their definitions come from their implementations :) ). Should I file a separate issue or can we repurpose this one? Of course this could be fixed in separate PRs.

**Comment by domenic** [comment]
We can repurpose this one to generally cover adding https://drafts.csswg.org/cssom-view/#extensions-to-the-mouseevent-interface to https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/events/MouseEvent.webidl and https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/events/MouseEvent-impl.js

**Comment by domenic** [comment]
It looks like this has been implemented for some time, I believe since https://github.com/jsdom/jsdom/pull/3484.
